### PR TITLE
Simplify skin bindable flow

### LINKED ISSUE #20234: Deleting a skin while in skin editor creates two default skins
### Type

Game behaviour

### Bug description

Deleting a skin while in the skin editor resets the skin and creates a new copy of the defaulted deleted skin

Steps to reproduce:

- Have a user made skin
- Open skin editor from settings
- Open the settings once again in the skin editor (Ctrl + O)
- Delete selected skin 

Now you have two default skins

### Screenshots or videos

https://user-images.githubusercontent.com/73758693/189520019-69f0bf63-f8e6-43dd-9baa-a30d23f2d284.mp4

### Version

2022.910.0-lazer

### Logs

[runtime.log](https://github.com/ppy/osu/files/9542142/runtime.log)
[network.log](https://github.com/ppy/osu/files/9542143/network.log)
[database.log](https://github.com/ppy/osu/files/9542144/database.log)
[performance.log](https://github.com/ppy/osu/files/9542145/performance.log)

### PR DESCRIPTION
**Pull request by peppy**
Now, nothing touches the configuration apart from `OsuGame`, making everything else flow better and avoid weird cyclic set bugs.

- [x] Depends on https://github.com/ppy/osu-framework/pull/5401
- [x] Depends on framework bump

Closes https://github.com/ppy/osu/issues/20234.

### CODE REVIEW

**Comment by peppy** [comment]
I recommend testing any edge case flows you can think of. I can't seem to break it but it's touching code which has historically been pretty delicate. In theory it shouldn't be delicate anymore but..

**Comment by frenzibyte** [comment]
The dropdown selection highlight no longer appears on initial load, reset to default, or picking skin via `<Random Skin>` or hotkey:

https://user-images.githubusercontent.com/22781491/189778727-55755b01-b683-4d52-bede-932599938c10.mp4

File: osu.Game/Overlays/Settings/Sections/SkinSection.cs [hunk 1]
```diff
@@ -73,47 +70,30 @@ private void load(OsuConfigManager config, [CanBeNull] SkinEditorOverlay skinEdi
...
         }
 
         protected override void LoadComplete()
         {
             base.LoadComplete();
 
-            skinDropdown.Current = dropdownBindable;
+            skinDropdown.Current = skins.CurrentSkinInfo;
 
             realmSubscription = realm.RegisterForNotifications(_ => realm.Realm.All<SkinInfo>()
```
Comment (frenzibyte) [inline osu.Game/Overlays/Settings/Sections/SkinSection.cs:81]: It's already specified in BDL:
```suggestion
            realmSubscription = realm.RegisterForNotifications(_ => realm.Realm.All<SkinInfo>()
```

**Review by frenzibyte — CHANGES_REQUESTED** [review]
See above.

**Comment by peppy** [comment]
Dropdown selection should now work correctly. But of a weird cause for this, but it's not the first time it's happened..

**Comment by smoogipoo** [comment]
The dropdown selection is still not working after game startup.

**Comment by peppy** [comment]
Hmm, that's due to setting `Current` before populating `Items`. An easy fix is to move the `Current` initialisation to inside the realm subscription, but I'll take a look at the dropdown logic to see if there's a better way to handle that.

**Comment by peppy** [comment]
Remaining issue will be resolved by https://github.com/ppy/osu-framework/pull/5401.

**Review by frenzibyte — APPROVED** [review]

**Comment by smoogipoo** [comment]
Can this one be merged without the dependency? It's just a visual thing right?

**Review by smoogipoo — APPROVED** [review]

**Comment by frenzibyte** [comment]
Have added it as part of the dependencies since merging this PR without is sort of regressing the dropdown's behaviour, and that a framework bump should be done before publishing a game release with this PR merged (would imagine the bug affecting up/down-arrow selection as well since the selected item is unknown).

**Pull request merged**